### DISCUSSION TITLE
How can I ignore/prevent selection of node if it is being dragged

### DISCUSSION THREAD

**Discussion by KumarilDave** [xyflow/xyflow#3242] [Q&A] [answered]
Hello friends,
I am trying to achieve an interaction where, I have a custom node called EmptyNode. What I want to do is separate select interaction and drag interaction. On drag and drop of the node, I will update the edge to connect to the new Node where as on select, user will be able to configure a new node type. Right now if I drag immediately I see the AddNodeContainer and moment I release, I see EmptyNodeContainer. I would not like to select if the use tried to drag. I am ok with the interaction if use has to click again to select. Any idea ? 

`const EmptyNode = ({ selected, dragging }: NodeProps) => {
  console.log(selected)
  console.log(dragging)
  return (
    <>
      <Handle
        type="target"
        position={Position.Left}
        style={{ visibility: 'hidden', left: '0px' }}
      />
      {selected && !dragging ? (
        <EmptyNodeContainer>
            .....
        </EmptyNodeContainer>
      ) : (
        <AddNodeContainer>
           .....
        </AddNodeContainer>
      )}
    </>
  )
}`

**Comment by bcakmakoglu** [comment, marked as answer]
There's a prop to control that called `selectNodesOnDrag` though this disables the selection of any node type while dragging.
